boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_ipcc.c

```diff
@@ -0,0 +1,183 @@
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/ipcc.h>
+#include <debug.h>
+
+#include <stm32wl5_ipcc.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Define default values for macros if they are not define in config */
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN1_RXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN1_RXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN1_TXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN1_TXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN2_RXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN2_RXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN2_TXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN2_TXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN3_RXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN3_RXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN3_TXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN3_TXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN4_RXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN4_RXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN4_TXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN4_TXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN5_RXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN5_RXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN5_TXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN5_TXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN6_RXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN6_RXBUF 0
+#endif
+
+#ifndef CONFIG_ARCH_BOARD_IPCC_CHAN6_TXBUF
+#define CONFIG_ARCH_BOARD_IPCC_CHAN6_TXBUF 0
+#endif
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int init_ipcc(int chan, size_t rxbuflen, size_t txbuflen);
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: init_ipcc
+ *
+ * Description:
+ *   Initializes IPCC channel with tx and rx buffer sizes. If ipcc is
+ *   unbuffered, rxbuflen and txbuflen are ignored.
+ *
+ * Input Parameters:
+ *   chan - channel number, indexed from 0
+ *   rxbuflen - size of rxbuffer for buffered transactions
+ *   txbuflen - size of txbuffer for buffered transactions
+ *
+ * Returned Value:
+ *   0 on success or -1 on errors.
+ *
+ ****************************************************************************/
+
+static int init_ipcc(int chan, size_t rxbuflen, size_t txbuflen)
+{
+  struct ipcc_lower_s *ipcc;
+  int ret;
+
+#ifndef CONFIG_IPCC_BUFFERED
+  UNUSED(rxbuflen);
+  UNUSED(txbuflen);
```

Comment:
Optional: can be placed together with `ret = ipcc_register(ipcc);` in the same `ifdef` block.